src/main/java/edu/hm/hafner/AbstractMavenLogParser.java

```diff
@@ -0,0 +1,18 @@
+package edu.hm.hafner;
+
+import edu.hm.hafner.analysis.parser.MavenConsoleParser;
+
+/**
+ * A parser subclass for MavenConsoleParser, AntJavacParser, JavaDocParser, JavacParser
+ *
+ * @author Jagruti Tiwari
+ */
+
+public class AbstractMavenLogParser extends MavenConsoleParser {
+
+    @Override
+    protected boolean isLineInteresting(final String line) {
```

Comment:
Hey @uhafner ,

For each of the parsers where should I write the code to check the following:

> 
>     * `AntJavacParser` should ignore all warnings, if a maven goal is set (in `isLineInteresting`). Otherwise proceed as normal.
> 
>     * `JavaDocParser`, `JavacParser` should ignore warnings, that are from a different maven plugin.
> 
>     * `MavenConsoleParser` should work unchanged, just the  goal and plugin detection will be from the parent class